### PR TITLE
Generate baselines

### PR DESCRIPTION
**Pull request by mgax**
Fixes https://github.com/eaudeweb/ozone/issues/1385

The functionality is available in admin under the "Ozone Tools" section:
Screenshot 2019-10-14 at 18 31 31

Click on "Generate diff":
Screenshot 2019-10-14 at 18 31 39

Review and click "Apply changes":
Screenshot 2019-10-14 at 18 30 59

See the green messages and be happy:
Screenshot 2019-10-14 at 18 31 11

### CODE REVIEW

File: ozone/templates/admin/ozone_tools/generate_baselines.html [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    {% endfor %}
+    </tbody>
+  </table>
+
+  <input type="hidden" name="step" value="apply">
+  <button type="submit">Apply changes</button>
+
+{% else %}
+  <input type="hidden" name="step" value="diff">
+  <button type="submit">Generate diff</button>
```
Comment (melish) [inline ozone/templates/admin/ozone_tools/generate_baselines.html]: Replace text with "Preview changes"

File: ozone/templates/admin/ozone_tools/generate_baselines.html [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        <td>{{ record.row.baseline_type }}</td>
+        <td>{{ record.row.group }}</td>
+        <td>{{ record.row.baseline }}</td>
+        <td>{{ record.new_value }}</td>
+      </tr>
+    {% endfor %}
+    </tbody>
+  </table>
+
+  <h2>Obsolete baselines ({{ diff.obsolete|length }})</h2>
```
Comment (melish) [inline ozone/templates/admin/ozone_tools/generate_baselines.html]: Replace with "To be deleted"

File: ozone/templates/admin/ozone_tools/generate_baselines.html [hunk 3]
```diff
@@ -0,0 +1,89 @@
+{% extends "admin/base_site.html" %}
+{% block content %}
+
+<form method="POST">
+{% csrf_token %}
+
+{% if step == "diff" %}
+  <h2>Missing baselines ({{ diff.missing|length }})</h2>
```
Comment (melish) [inline ozone/templates/admin/ozone_tools/generate_baselines.html]: Replace with "To be created"

File: ozone/templates/admin/ozone_tools/generate_baselines.html [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        <td>{{ record.party }}</td>
+        <td>{{ record.baseline_type }}</td>
+        <td>{{ record.group }}</td>
+        <td>{{ record.baseline }}</td>
+      </tr>
+    {% endfor %}
+    </tbody>
+  </table>
+
+  <h2>Different baselines ({{ diff.different|length }})</h2>
```
Comment (melish) [inline ozone/templates/admin/ozone_tools/generate_baselines.html]: Replace with "To be updated"

File: ozone/templates/admin/ozone_tools/generate_baselines.html [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+  <table>
+    <thead>
+      <th><input type="checkbox" disabled></th>
+      <th>Party</th>
+      <th>Baseline type</th>
+      <th>Substance group</th>
```
Comment (melish) [inline ozone/templates/admin/ozone_tools/generate_baselines.html]: Replace with "Annex/Group"

**Comment by mgax** [comment]
Turns out, I broke the management command, because of the `yield`. Working on refactoring.

**Comment by mgax** [comment]
> Turns out, I broke the management command, because of the `yield`. Working on refactoring.

And done. The loop is simple enough, no point in butchering the management command.

@melish should I also move `get_baseline` and its dependencies to `ozone/core/calculated/baselines.py`? It feels like a more natural home for them. The management command should, in theory, call reusable business logic from elsewhere, not implement valuable logic by itself.

**Pull request closed without merging**